src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonfwJavaProfile.java

```diff
@@ -0,0 +1,114 @@
+package com.devonfw.ide.sonarqube.common.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.plugins.java.Java;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * This class creates a quality profile containing the rules of this plugin plus additional rules from external repos.
+ */
+@SonarLintSide
+public class DevonfwJavaProfile implements BuiltInQualityProfilesDefinition {
+
+  private static final Set<String> FORBIDDEN_RULE_KEYS = new HashSet<>(Arrays.asList(
+      /* squid repo */
+      "S2076", "S2078", "S3318", "S2070", "S4142",
+      /* findbugs repo */
+      "VA_FORMAT_STRING_ARG_MISMATCH", "VA_FORMAT_STRING_MISSING_ARGUMENT", "VA_FORMAT_STRING_EXTRA_ARGUMENTS_PASSED",
+      "VA_FORMAT_STRING_BAD_CONVERSION_FROM_ARRAY", "VA_FORMAT_STRING_BAD_ARGUMENT",
+      "VA_FORMAT_STRING_BAD_CONVERSION_TO_BOOLEAN", "VA_FORMAT_STRING_NO_PREVIOUS_ARGUMENT", "VA_FORMAT_STRING_ILLEGAL",
+      "VA_FORMAT_STRING_BAD_CONVERSION", "VA_FORMAT_STRING_EXPECTED_MESSAGE_FORMAT_SUPPLIED",
+      /* fb-contrib repo */
+      "SPP_NULL_CHECK_ON_MAP_SUBSET_ACCESSOR", "SPP_NULL_CHECK_ON_OPTIONAL", "SPP_USE_CONTAINSKEY"));
```

Comment:
Have these rules been removed? Then why dont we just remove them from the XML file?

Reply:
Partly removed or merged with other rules, partly moved to the paid SonarQube version.
I didn't want to remove anything from the XML yet, but if it's fine I will do it.